File: javascript/rna-transcription/rna-transcription.js
let complements = {G: "C", C: "G", T: "A", A: "U"}

function DnaTranscriber() {}

DnaTranscriber.prototype.toRna = function (dna) {
  return dna.split("").map(nucleotide => complements[nucleotide]).join("")
}

module.exports = DnaTranscriber

File: ocaml/rna-transcription/rna_transcription.ml
type dna = [ `A | `C | `G | `T ]
type rna = [ `A | `C | `G | `U ]

let convert nucleotide = match nucleotide with
  | `G -> `C
  | `C -> `G
  | `T -> `A
  | `A -> `U

let to_rna dna = List.map (convert) dna
